File: software/examples/java/ExampleFindCorners.java
import com.tinkerforge.BrickletJoystick;
import com.tinkerforge.IPConnection;

public class ExampleFindCorners {
	private static final String host = "localhost";
	private static final int port = 4223;
	private static final String UID = "ABC"; // Change to your UID
	
	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletJoystick joy = new BrickletJoystick(UID, ipcon); // Create device object

		ipcon.connect(host, port); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get threshold callbacks with a debounce time of 0.2 seconds (200ms)
		joy.setDebouncePeriod(200);

		// Configure threshold for "x and y value outside of -99 and 99"
		joy.setPositionCallbackThreshold('o', (short)-99, (short)99, (short)-99, (short)99);

		// Add and implement position reached listener 
		// (called if x and y value outside of -99 and 99)
		joy.addPositionReachedListener(new BrickletJoystick.PositionReachedListener() {
			public void positionReached(short x, short y) {
				if(x == 100 && y == 100) {
					System.out.println("Top Right");
				} else if(x == -100 && y == 100) {
					System.out.println("Top Left");
				} else if(x == -100 && y == -100) {
					System.out.println("Bottom Left");
				} else if(x == 100 && y == -100) {
					System.out.println("Bottom Right");
				} else {
					// This can't happen, the threshold is configured to: "outside of -99, 99"
					System.out.println("Error");
				}
			}
		});

		System.console().readLine("Press key to exit\n");
	}
}

File: software/examples/csharp/ExampleFindCorners.cs
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "ABC"; // Change to your UID

	// Callback for x and y position outside of -99, 99
	static void ReachedCB(BrickletJoystick sender, short x, short y)
	{
		if(x == 100 && y == 100)
		{
			System.Console.WriteLine("Top Right");
		}
		else if(x == -100 && y == 100)
		{
			System.Console.WriteLine("Top Left");
		}
		else if(x == -100 && y == -100)
		{
			System.Console.WriteLine("Bottom Left");
		}
		else if(x == 100 && y == -100)
		{
			System.Console.WriteLine("Bottom Right");
		}
		else
		{
			// This can't happen, the threshold is configured to: "outside of -99, 99"
			System.Console.WriteLine("Error");
		}
	}

	static void Main() 
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletJoystick joy = new BrickletJoystick(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get threshold callbacks with a debounce time of 0.2 seconds (200ms)
		joy.SetDebouncePeriod(200);

		// Register threshold reached callback to function ReachedCB
		joy.PositionReached += ReachedCB;

		// Configure threshold for "x and y value outside of -99 and 99"
		joy.SetPositionCallbackThreshold('o', -99, 99, -99, 99);

		System.Console.WriteLine("Press key to exit");
		System.Console.ReadKey();
	}
}
